### PR TITLE
`Development`: Add documentation for learning paths

### PR DESCRIPTION
**Pull request by MaximilianAnzinger**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### CODE REVIEW

File: docs/user/adaptive-learning.rst [hunk 1]
```diff
@@ -64,6 +67,35 @@ Alternatively, instructors can also link competencies to an exercise or lecture
...
+Recommendations (visualized on the left) are generated via an intelligent agent that accounts for multiple metrics, e.g. prior performance, confidence, relations, and due dates, to support students in their selection of learning resources.
+Students can use the up and down buttons to navigate to the previous or next recommendation respectively. Hovering over a node in the list will display more information about the learning resource.
+
+|students-learning-path-participation|
+
+Students can access their learning path graph via the eye icon on the top left. The graph displays all competencies, lecture units, exercises, and their relations. Each competency consists of a start node, visualized by the competency rings displaying progress, confidence, and overall mastery, and an end node represented by a checkered flag. Learning resources are linked to a competency via edges tot he respective start and end nodes. If the resource is still pending, it is represented as a play symbol. Once the task is completed, it is visualized as a checkmark.
```
Comment (AntonGeTUM) [inline docs/user/adaptive-learning.rst:82]: "via edges _tot he_ respective start and end nodes" -> "via edges _to the_ respective start and end nodes"

File: docs/user/adaptive-learning.rst [hunk 2]
```diff
@@ -64,6 +67,35 @@ Alternatively, instructors can also link competencies to an exercise or lecture
...
+Users can read the graph from top to bottom starting with the competencies that have no prerequisites, continuing downwards towards competencies that build up on the prior knowledge. Students can zoom, pan, and drag the graph to navigate. For better orientation, the top right corner contains a mini-map.
```
Comment (AntonGeTUM) [inline docs/user/adaptive-learning.rst:83]: "Users can read the graph from top to bottom starting with the competencies" -> "Users can read the graph from top to bottom, starting with the competencies"

"competencies that build _up on the_ prior knowledge" -> "competencies that build _upon/on_ prior knowledge"

File: docs/user/adaptive-learning.rst [hunk 3]
```diff
@@ -64,6 +67,35 @@ Alternatively, instructors can also link competencies to an exercise or lecture
...
+On the bottom right of the graph, users can view a legend describing the different types of nodes.
+Hovering over any node, i.e. exercise or competency, opens a popover containing essential information about the item, e.g. the type of exercise and title, or for competencies the details including the description.
+
+|students-learning-path-graph|
+
+Learning Paths for Instructors
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Instructors can enable learning paths for their course either by editing the course or on the dedicated learning path management page. This will generate individualized learning paths for all course participants.
```
Comment (Ceiridge) [inline docs/user/adaptive-learning.rst:92]: Better: learning path for their courses

File: docs/user/adaptive-learning.rst (hunk 2 above)
Comment (Ceiridge) [inline docs/user/adaptive-learning.rst:83]: _towards_ is informal, _toward_ is formal and thus better

File: docs/user/adaptive-learning.rst [hunk 4]
```diff
@@ -9,10 +9,13 @@ Adaptive Learning
...
+
+Competencies
+------------
 
 Competencies for Students
--------------------------
+^^^^^^^^^^^^^^^^^^^^^^^^^
 Competencies allow students to understand the learning goals of the course and measure their progress towards achieving them.
 
 In Artemis, a competency is considered mastered by a student when they completed all linked learning objects (progress equals 100%) and have an adequate confidence level (average score greater or equal to mastery threshold set by the instructor).
```
Comment (Ceiridge) [inline docs/user/adaptive-learning.rst:21]: Change this to `when they complete`. Otherwise this is incorrect grammar

Comment (Ceiridge) [inline docs/user/adaptive-learning.rst:21]: Also change this to `equal to the mastery threshold set`, as it refers to a specific one

**Review by Ceiridge — CHANGES_REQUESTED** [review]
Looks good in total, but there are some minor mistakes

File: docs/user/adaptive-learning.rst [hunk 5]
```diff
@@ -64,6 +67,35 @@ Alternatively, instructors can also link competencies to an exercise or lecture
...
+Hovering over any node, i.e. exercise or competency, opens a popover containing essential information about the item, e.g. the type of exercise and title, or for competencies the details including the description.
+
+|students-learning-path-graph|
+
+Learning Paths for Instructors
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Instructors can enable learning paths for their course either by editing the course or on the dedicated learning path management page. This will generate individualized learning paths for all course participants.
+
+Once the feature is enabled, instructors get access to each student's learning path. Instructors can search for students by name or login and view their respective learning path graph.
```
Comment (LenaKahle) [inline docs/user/adaptive-learning.rst:94]: "Instructors can search for students by name or login [...]" -> Does login mean the ge12abc username? It is a bit ambiguous and could also be interpreted as "Instructors can login as the student and view the graph"

**Review by LenaKahle — CHANGES_REQUESTED** [review]
Looks good, I added a small comment

**Review by egekurt123 — DISMISSED** [review]
Documentation text LGTM

File: docs/user/adaptive-learning.rst [hunk 6]
```diff
@@ -64,6 +67,35 @@ Alternatively, instructors can also link competencies to an exercise or lecture
...
+
+Students can access their learning path graph via the eye icon on the top left. The graph displays all competencies, lecture units, exercises, and their relations. Each competency consists of a start node, visualized by the competency rings displaying progress, confidence, and overall mastery, and an end node represented by a checkered flag. Learning resources are linked to a competency via edges tot he respective start and end nodes. If the resource is still pending, it is represented as a play symbol. Once the task is completed, it is visualized as a checkmark.
+Users can read the graph from top to bottom starting with the competencies that have no prerequisites, continuing downwards towards competencies that build up on the prior knowledge. Students can zoom, pan, and drag the graph to navigate. For better orientation, the top right corner contains a mini-map.
+On the bottom right of the graph, users can view a legend describing the different types of nodes.
+Hovering over any node, i.e. exercise or competency, opens a popover containing essential information about the item, e.g. the type of exercise and title, or for competencies the details including the description.
```
Comment (JohannesStoehr) [inline docs/user/adaptive-learning.rst:85]:
```suggestion
Hovering over any node, e.g. exercise or competency, opens a popover containing essential information about the item, e.g. the type of exercise and title, or for competencies the details including the description.
```
You don't mention lecutres here, so they are only examples

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]

File: docs/user/adaptive-learning.rst [hunk 7]
```diff
@@ -64,6 +67,35 @@ Alternatively, instructors can also link competencies to an exercise or lecture
 
 |instructors-learning-goals-link|
 
+Learning Paths
+--------------
+
+Learning Paths for Students
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Students can access their learning path in the learning path tab. Here they are able to access recommended lecture units and participate in exercises.
```
Comment (JanaNF) [inline docs/user/adaptive-learning.rst:76]: Missing comma after "Here they are..." --> "Here, they are..."
I would change "Here they are able to... " --> "Here, they can ..."

File: docs/user/adaptive-learning.rst (hunk 1 above)
Comment (JanaNF) [inline docs/user/adaptive-learning.rst:82]: The sentences "Learning resources are linked ... ", "If the resources is still ..." and "Once the task is completed, ..." are written in the passive voice. Please consider rephrasing them to active.

**Review by JanaNF — CHANGES_REQUESTED** [review]
I checked with Grammarly, and I would incorporate the following suggestions/comments.

File: docs/user/adaptive-learning.rst (hunk 5 above)
Comment (MaximilianAnzinger) [reply docs/user/adaptive-learning.rst:94]: Other universities don't use LRZ-Kennung.

Would `login credentials` work better? However, credentials would include the password ...

File: docs/user/adaptive-learning.rst (hunk 3 above)
Comment (MaximilianAnzinger) [reply docs/user/adaptive-learning.rst:92]: I don't quite understand what you mean. Could you suggest the change using the Github feature?

File: docs/user/adaptive-learning.rst [hunk 8]
```diff
@@ -64,6 +67,35 @@ Alternatively, instructors can also link competencies to an exercise or lecture
...
+
+Students can access their learning path graph via the eye icon on the top left. The graph displays all competencies, lecture units, exercises, and their relations. Each competency consists of a start node, visualized by the competency rings displaying progress, confidence, and overall mastery, and an end node represented by a checkered flag. Edges link learning resources to a competency via the respective start and end nodes. If the resource is still pending, it displays as a play symbol. Upon completion of the task, it appears as a checkmark.
+Users can read the graph from top to bottom, starting with the competencies that have no prerequisites, continuing downwards toward competencies that build upon prior knowledge. Students can zoom, pan, and drag the graph to navigate. For better orientation, the top right corner contains a mini-map.
+On the bottom right of the graph, users can view a legend describing the different types of nodes.
+Hovering over any node, e.g. exercise or competency, opens a popover containing essential information about the item, e.g. the type of exercise and title, or for competencies the details including the description.
```
Comment (sarpsahinalp) [inline docs/user/adaptive-learning.rst:85]: For better understanding "or for competencies the details including the description" --> "or for competencies, the details, including the description."

File: docs/user/adaptive-learning.rst [hunk 9]
```diff
@@ -64,6 +67,35 @@ Alternatively, instructors can also link competencies to an exercise or lecture
...
+
+|students-learning-path-graph|
+
+Learning Paths for Instructors
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Instructors can enable learning paths for their course either by editing the course or on the dedicated learning path management page. This will generate individualized learning paths for all course participants.
```
Comment (sarpsahinalp) [inline docs/user/adaptive-learning.rst:92]: "learning paths for their course" --> "learning paths for their courses"

**Review by sarpsahinalp — COMMENTED** [review]
Grammerly suggested following changes for a better understanding.

File: docs/user/adaptive-learning.rst (hunk 5 above)
Comment (LenaKahle) [reply docs/user/adaptive-learning.rst:94]: "Login username" or "handle" could work.

**Review by LenaKahle — APPROVED** [review]
Approve, looks good.

**Review by JohannesStoehr — APPROVED** [review]

**Review by egekurt123 — APPROVED** [review]

**Review by xHadie — APPROVED** [review]
LGTM!
Reads  well and is easy to understand.

**Comment by MaximilianJG** [comment]
Documentation LGTM. Very clear and well written!

**Review by vinceclifford — APPROVED** [review]
Easy to follow and to understand

**Review by Predixx — APPROVED** [review]
Looks good to me

**Pull request merged**